### PR TITLE
Re-enable Packaging Tests

### LINKED ISSUE #4341: re-enable native image tests in pipeline on master

### PR DESCRIPTION
**Pull request by tomas-langer**
Resolves #4341 for jar, jlink and native image for SE, MP will be fixed in a follow up PR

As part of this change, we had to switch to Parsson JSON-P implementation. This requires multiple changes in dependencies.

### CODE REVIEW

File: microprofile/cdi/src/main/java/module-info.java [hunk 1]
```diff
@@ -23,6 +23,8 @@
     // needed for Unsafe used from Weld
     requires jdk.unsupported;
     requires java.logging;
+    // weld requires java.sql.Data and we fail if not on classpath
```
Comment (ljnelson) [inline microprofile/cdi/src/main/java/module-info.java:26]: Hmm; no such class. Do you mean `java.sql.Date`?

**Review by barchetta — APPROVED** [review]
Approve dependency changes.

File: microprofile/cdi/src/main/java/module-info.java (hunk 1 above)
Comment (tomas-langer) [reply microprofile/cdi/src/main/java/module-info.java]: yes

**Comment by tomas-langer** [comment]
Merging even though there is a failing test - the test failure is intermittent and known

**Pull request merged**